qemu/tests/pci_hotplug.py

```diff
@@ -92,6 +92,16 @@ def verify_supported_device(dev):
             raise error.TestError("%s doesn't support device: %s" %
                                   (cmd_type, dev))
 
+    def device_add_balloon(pci_num, queues=1):
+        device_id = pci_type + "-" + utils_misc.generate_random_id()
+        pci_info.append([device_id, device_id])
+
+        pci_model = params.get("pci_model")
+        verify_supported_device(pci_model)
+        pci_add_cmd = "device_add id=%s,driver=%s" % (pci_info[pci_num][1],
+                                                      pci_model)
+        return device_add(pci_num, pci_add_cmd)
+
```

Comment:
Hi coli,

suggest to separate  balloon hotplug, pci_hotplug will be longer and longer,
or you can put it in qemu.py like memory hotplug